cunumeric/deferred.py

```diff
@@ -1596,7 +1616,9 @@ def tile(self, rhs: Any, reps: Union[Any, Sequence[int]]) -> None:
         task.execute()
 
     # Transpose the matrix dimensions
-    def transpose(self, axes: Any) -> DeferredArray:
+    def transpose(
+        self, axes: Union[None, tuple[int, ...], list[int]]
```

Comment:
Will `Union[Sequence[int], None]` suffice? Sometime soon it would be good to survey all the various "axis"/"axes" types and see if there are some aliases we can define.

Reply:
On the "eager" version of this function https://github.com/nv-legate/cunumeric/blob/branch-22.07/cunumeric/eager.py#L494 we pass the axes straight to the NumPy transpose function, which according to https://numpy.org/doc/stable/reference/generated/numpy.transpose.html only supports lists and tuples. We could instead accept any `Sequence` and cast it to a tuple before passing to NumPy.

Actually, it looks like the "deferred" version of this is even more strict, since we pass the axes to `Store.transform` https://github.com/nv-legate/cunumeric/blob/branch-22.07/cunumeric/deferred.py#L1600, which only accepts tuples https://github.com/nv-legate/legate.core/blob/branch-22.07/legate/core/store.py#L1162. Weird that mypy didn't complain.